features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/RoomMemberListView.kt

```diff
@@ -116,9 +124,53 @@ fun RoomMemberListView(
     }
 }
 
+@Composable
+private fun RoomMemberList(
+    roomMembers: Async.Success<RoomMembers>,
+    onUserSelected: (RoomMember) -> Unit,
+) {
+    LazyColumn(modifier = Modifier.fillMaxWidth(), state = rememberLazyListState()) {
+        if (roomMembers.state.invited.isNotEmpty()) {
+            item {
+                Text(
+                    modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
+                    text = stringResource(id = R.string.screen_room_member_list_pending_header_title),
+                    style = ElementTextStyles.Regular.callout,
+                    color = MaterialTheme.colorScheme.secondary,
+                    textAlign = TextAlign.Start,
+                )
+            }
+            items(roomMembers.state.invited) { matrixUser ->
+                RoomMemberSearchResultItem(
+                    modifier = Modifier.fillMaxWidth(),
+                    roomMember = matrixUser,
+                    onClick = { onUserSelected(matrixUser) }
+                )
+            }
+        }
+        item {
+            val memberCount = roomMembers.state.joined.count()
+            Text(
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
+                text = pluralStringResource(id = R.plurals.screen_room_member_list_header_title, count = memberCount, memberCount),
+                style = ElementTextStyles.Regular.callout,
+                color = MaterialTheme.colorScheme.secondary,
+                textAlign = TextAlign.Start,
+            )
+        }
+        items(roomMembers.state.joined) { matrixUser ->
+            RoomMemberSearchResultItem(
+                modifier = Modifier.fillMaxWidth(),
+                roomMember = matrixUser,
+                onClick = { onUserSelected(matrixUser) }
+            )
+        }
```

Comment:
this code is almost the same as in the previous block with different data & header, it would make sense to factorize it to a dedicated composable with some parameters, wdyt?
e.g.
```
@Composable
fun RoomMemberSection(
    title: String,
    members: ImmutableList<RoomMember>,
    onMemberSelected: (RoomMember) -> Unit, 
    modifier: Modifier = Modifier,
}
```

Reply:
I like the idea but it gets a bit "weird" because we are inside a `LazyListScope` which is not a composable function.

This is how it would turn out, let me know what you think (I'd like your opinion too @jmartinesp ):

```
@Composable
private fun RoomMemberList(
    roomMembers: Async.Success<RoomMembers>,
    onUserSelected: (RoomMember) -> Unit,
) {
    LazyColumn(modifier = Modifier.fillMaxWidth(), state = rememberLazyListState()) {
        if (roomMembers.state.invited.isNotEmpty()) {
            roomMemberListSection(
                headerText = { stringResource(id = R.string.screen_room_member_list_pending_header_title) },
                members = roomMembers.state.invited,
                onMemberSelected = { onUserSelected(it) }
            )
        }
        val memberCount = roomMembers.state.joined.count()
        roomMemberListSection(
            headerText = { pluralStringResource(id = R.plurals.screen_room_member_list_header_title, count = memberCount, memberCount) },
            members = roomMembers.state.joined,
            onMemberSelected = { onUserSelected(it) }
        )
    }
}

private fun LazyListScope.roomMemberListSection(
    headerText: @Composable () -> String,
    members: ImmutableList<RoomMember>,
    onMemberSelected: (RoomMember) -> Unit,
) {
    item {
        Text(
            modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
            text = headerText(),
            style = ElementTextStyles.Regular.callout,
            color = MaterialTheme.colorScheme.secondary,
            textAlign = TextAlign.Start,
        )
    }
    items(members) { matrixUser ->
        RoomMemberSearchResultItem(
            modifier = Modifier.fillMaxWidth(),
            roomMember = matrixUser,
            onClick = { onMemberSelected(matrixUser) }
        )
    }
}
```